# Chapter 02

## 02.03 System of Units

A system of units has the following components: 
- Base units - for length, mass, time, temperature, electrical current, and light intensity 
- Multiple units - multiples or fractions of base units (i. e. minutes, hours, seconds all different ways to express time) 3. Derived units - Two ways to obtain:

(a) compound units - multiplying and dividing base or multiple units (i. e. $cm^2$, ft/min) 

(b) defined equivalents of comound units (i. e. 1 erg $\equiv$ (1g cm/$s^2$)) 

**Systeme Internationale d'Unites (SI)** - system of units with widespread acceptance:
- length -> meter (m) 
- mass -> kilogram (kg) 
- time -> seconds (s) 

SI uses prefixes to indicate powers of ten. For example:
- mega (M) = 10^6 
- kilo (k) = 10^3 
- centi (c) = 10^-2 
- mili (m) = 10^-3 
- micro ($\mu$) = 10^-6 

**CGS system** is like SI except:
- mass -> grams (g) 
- length -> cenimeters (cm) 

**American engineering system** has two issues: 
conversions are not powers of 10 and unit of force (discussed next section):
- length -> foot (ft) 
- mass -> pound-mass ($lb_m$) 
- time -> second (s) 

Conversion between systems is possible using the correct factors. See table 2.3-1 or consult Perry's Chemical Engineers' Handbook for a complete list

### Test Yourself
#### Problem 1
What are the factors (numerical values and units) needed to convert: 

- meters to milimeters
- nanoseconds to second 
- square centimeters to square meters 
- cubic feet to cubic meters 
- horsepower to British thermal units per second


In [9]:
import pint
ureg = pint.UnitRegistry()

print((1 *  ureg.meter).to(ureg.millimeter))
print((1 * ureg.nanosecond).to(ureg.second))
print((1 * ureg.centimeter**2).to(ureg.meter**2))
print((1 * ureg.foot**3).to(ureg.meter**3))
print((1 * ureg.hp).to(ureg.BTU / ureg.second))

1000.0 millimeter
1e-09 second
0.0001 meter ** 2
0.028316846591999994 meter ** 3
0.7067871049011177 btu / second


#### Problem 2
What is the derived SI unit for velocity? The velocity unit in the CGS system? In the American engineering system?

Velocity is [length]/[time]:
- SI  -> m/s
- CGF -> cm/s
- AES -> ft/s

### Example 2-3.1
#### Conversion between System of Units

Convert 23 $lb_mft/min^2$ to its equivalent $kg-cm/s^2$

Write dimensional equation, fill in units of conversion factors (new/old) and then the numerical values of these factors. Finally, do arithmetic

In [2]:
import pint
ureg = pint.UnitRegistry()

print((1 * ureg.pound).to(ureg.kilogram))
print((1 * ureg.foot).to(ureg.meter))
print((1 * ureg.meter).to(ureg.centimeter))
print((1 * ureg.minute).to(ureg.second))

0.45359237 kilogram
0.30479999999999996 meter
100.0 centimeter
60.0 second


Setup equation with correct orientation

$$ 
\frac{23 lb-ft^2}{1 min^2}
\frac{0.45359237 kg}{1 lb}
\frac{1 m}{3.281 ft}
\frac{100 cm}{1 m}
\frac{1 min^2}{60^2 s^2}
$$

In [3]:
(23 * ureg.pound * ureg.foot / ureg.minute**2).to(
      ureg.kilogram * ureg.centimeter / ureg.second**2)

<Quantity(0.08832955418466666, 'centimeter * kilogram / second ** 2')>